hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -66,26 +75,31 @@
   private ExecutorService executorService;
   private TServer server;
   private HiveMetaStore.HMSHandler baseHandler;
+  private IMetaStoreClient client = null;
 
   public void start() {
     try {
-      hiveLocalDir = createTempDirectory("hive", asFileAttribute(fromString("rwxrwxrwx"))).toFile();
+      this.hiveLocalDir = createTempDirectory("hive", asFileAttribute(fromString("rwxrwxrwx"))).toFile();
       File derbyLogFile = new File(hiveLocalDir, "derby.log");
       System.setProperty("derby.stream.error.file", derbyLogFile.getAbsolutePath());
       setupMetastoreDB("jdbc:derby:" + getDerbyPath() + ";create=true");
 
       TServerSocket socket = new TServerSocket(0);
       int port = socket.getServerSocket().getLocalPort();
-      hiveConf = newHiveConf(port);
-      server = newThriftServer(socket, hiveConf);
-      executorService = Executors.newSingleThreadExecutor();
-      executorService.submit(() -> server.serve());
+      this.hiveConf = newHiveConf(port);
+      this.server = newThriftServer(socket, hiveConf);
+      this.executorService = Executors.newSingleThreadExecutor();
+      this.executorService.submit(() -> server.serve());
+      this.client = new HiveMetaStoreClient(hiveConf);
```

Comment:
I'd recommend taking a look at other classes based on this and copying their use of a client pool, rather than creating a client here.

Reply:
This might fix the JDK 11 tests, which appear to be failing because of connections to the Hive metastore. My guess is that the "write error" is that it times out because the metastore ran out of thread -- and we don't want to increase those threads.

JDK 8 tests are failing because of a constructor is being used that accepts a HiveConf. That needs to be done using reflection now.

Reply:
Using a clientPool fixed these issues, but it seems that we do not have enough worker in the HMS and this causes failing tests. Need to investigate why the  clients are not closed.

Thanks for taking a look!

Reply:
I know that it's annoying to track down, but that's exactly why we want to limit the worker threads. Much better to catch leaked connections here rather than trying to fix them for customers at runtime!

Reply:
This was fun 😄 
Using System.properties to configure the HMS made the ClientPool use the wrong URI.
Moved every metastore related stuff to the TestMetaStore